src/Processors/QueryPlan/DistinctStep.cpp

```diff
@@ -103,7 +103,12 @@ void DistinctStep::transformPipeline(QueryPipelineBuilder & pipeline, const Buil
                             return nullptr;
 
                         return std::make_shared<DistinctSortedChunkTransform>(
-                            header, set_size_limits, limit_hint, distinct_sort_desc, columns, false);
+                            header,
+                            set_size_limits,
+                            limit_hint,
+                            distinct_sort_desc,
+                            columns,
+                            input_stream.sort_scope == DataStream::SortScope::Stream);
```

Comment:
Hm. Now I think it's a bit strange to have a flag `sorted_stream` which only changes the transform name. What's the reason if everything else is same?

Reply:
IMO, it provides clearer message when looking at pipeline if distinct implementation is applied to sorted stream or just sorted chunks.